xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -67,37 +79,36 @@ public XdsServerBuilder xdsServingStatusListener(
     return this;
   }
 
-  /**
-   * Unsupported call. Users should only use {@link #forPort(int, ServerCredentials)}.
-   */
-  @DoNotCall("Unsupported. Use forPort(int, ServerCredentials) instead")
-  public static ServerBuilder<?> forPort(int port) {
-    throw new UnsupportedOperationException(
-        "Unsupported call - use forPort(int, ServerCredentials)");
+  public static XdsServerBuilder forPort(int port) {
+    return forPort(port, InsecureServerCredentials.create());
   }
 
   /** Creates a gRPC server builder for the given port. */
   public static XdsServerBuilder forPort(int port, ServerCredentials serverCredentials) {
-    NettyServerBuilder nettyDelegate = NettyServerBuilder.forPort(port, serverCredentials);
+    checkNotNull(serverCredentials, "serverCredentials");
+    InternalProtocolNegotiator.ServerFactory originalNegotiatorFactory =
+            InternalNettyServerCredentials.toNegotiator(serverCredentials);
+    ServerCredentials wrappedCredentials = InternalNettyServerCredentials.create(
```

Comment:
The `wrappedCredentials` is there only to insert the `FilterChainMatchingNegotiator` in the chain but does not really have any security/credentials functionality. If we can find a way to pass a "wrapping" `ProtocolNegotiator.ServerFactory` directly to the `NettyServerBuilder` we won't need the pseudo `wrappedCredentials`.

Reply:
Yeah, I think wrapping the credential like this is pretty fair. It is an abuse of ServerCredentials, but there's already other cases where the difference between "credentials" and "handshakers" is muddied and they haven't caused any real harm. This is internal, so we can always improve it later if it does become a problem.